python/TestHarness/resultsreader/results.py

```diff
@@ -0,0 +1,382 @@
+
+    @property
+    def pr_num(self) -> Union[int, None]:
+        """
+        Get the PR number associated with the test (if any)
+        """
+        assert(self.data['pr_num'] == self.results.pr_num)
+        return self.data['pr_num']
+
+    @property
+    def base_sha(self) -> Union[str, None]:
+        """
+        Get the base commit that these tests were ran on
+        """
+        value = None
+        if self.results.civet_version > 1:
+            value = self.data['base_sha']
+        assert(value == self.results.base_sha)
+        return value
+
+    @property
+    def time(self) -> datetime:
+        """
+        Get the time this test was added to the database
+        """
+        return self.data['time']
+
+    @property
+    def tester(self) -> str:
+        """
+        Get the Tester entry in the data
+        """
+        return self.data['tester']
+
+    @property
+    def json_metadata(self) -> dict:
+        """
+        Get the 'json_metadata' entry from the Tester entry
+        """
+        return self.tester.get('json_metadata', {})
+
+    @property
+    def hpc(self) -> dict:
+        """
+        Get the 'hpc' entry if it exists (otherwise, empty dict)
+        """
+        return self.data.get('hpc', {})
+
+    @property
+    def hpc_id(self) -> Union[int, None]:
+        """
+        Get the HPC job ID that ran this this test, if any
+        """
+        return self.hpc.get('id')
+
+    @property
+    def validation(self) -> dict:
+        """
+        Get the 'validation' entry for the test, if any
+        (otherwise, empty dict)
+
+        Contains the validation data and results
+        """
+        return self.data.get('validation', {})
+
+    @property
+    def validation_data(self) -> dict[str, ValidationData]:
+        """
+        Get the 'data' entry in 'validation' for this test, if any
+        (otherwise, empty dict)
+        """
+        return self._validation_data
+
+    @property
+    def validation_results(self) -> list[ValidationResult]:
+        """
+        Get the 'data' entry in 'validation' for this test, if any
+        (otherwise, empty dict)
+        """
+        return self._validation_results
+
+    def _buildValidationResults(self) -> list[ValidationResult]:
+        """
+        Internal method (use on construction) for converting validation
+        results in JSON to the underlying ValidationResult objects
+        """
+        return [ValidationResult(**v) for v in self.validation.get('results', [])]
+
+    def _buildValidationData(self) -> dict[str, ValidationData]:
+        """
+        Internal method (use on construction) for converting validation
+        data in JSON to the underlying ValidationData objects
+        """
+        input = self.validation.get('data', {})
+        data = {}
+
+        for k, v in input.items():
+            # Bounds is saved as a list; convert to tuple
+            if 'bounds' in v:
+                v['bounds'] = tuple(v['bounds'])
```

Comment:
Slight micro-optimization since this is in a loop:
```suggestion
            try:
                v['bounds'] = tuple(v['bounds'])
            except KeyError:
                pass
```

Reply:
Is catching an exception not slower?